File: src/cms/server1/cms/src/main/java/cn/javat/csa/cms/dto/SysUserDTO.java
package cn.javat.csa.cms.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 系统用户 DTO
 */
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Accessors(chain = true)
public class SysUserDTO implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 用户 ID
     */
    private Long id;

    /**
     * 用户名（登录账号）
     */
    private String username;

    /**
     * 昵称
     */
    private String nickname;

    /**
     * 头像 URL
     */
    private String avatar;

    /**
     * 邮箱
     */
    private String email;

    /**
     * Steam 好友 ID
     */
    private Long steamId;

    /**
     * 用户角色 0 一般用户 1 管理员
     */
    private Integer role;

    /**
     * 是否禁用 0 FALSE 1 TRUE
     */
    private Boolean isDisable;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    private LocalDateTime updateTime;
}

File: src/cms/front/src/types/SysUserDTO.ts
/**
 * 用户相关类型定义
 * User Type Definitions
 */

import { UserRole } from '@/enums/common'

/**
 * 用户信息 DTO
 * 对应后端：SysUserDTO
 */
export interface SysUserDTO {
    /** 用户 ID */
    id?: number
    /** 用户名 */
    username?: string
    /** 昵称 */
    nickname?: string
    /** 头像 URL */
    avatar?: string
    /** 邮箱 */
    email?: string
    /** Steam ID */
    steamId?: number
    /** 角色代码 */
    role?: UserRole
    /** 是否禁用 */
    isDisable?: boolean
    /** 创建时间 */
    createTime?: string
    /** 更新时间 */
    updateTime?: string
}

/**
 * 用户注册请求 DTO
 */
export interface UserRegisterDTO {
    /** 用户名 */
    username: string
    /** 密码 */
    password: string
    /** 确认密码 */
    confirmPassword?: string
    /** 昵称（可选） */
    nickname?: string
    /** 邮箱（可选） */
    email?: string
    /** 头像 URL */
    avatar?: string
}

/**
 * 用户登录请求 DTO
 */
export interface UserLoginDTO {
    /** 用户名或邮箱 */
    username: string
    /** 密码 */
    password: string
}

/**
 * 登录响应 DTO
 */
export interface LoginResponseDTO {
    /** JWT Token */
    token: string
    /** 用户信息 */
    user: SysUserDTO
}

/**
 * 更新用户信息请求 DTO
 */
export interface UpdateUserDTO {
    /** 昵称 */
    nickname?: string
    /** 头像 URL */
    avatar?: string
    /** 邮箱 */
    email?: string
}
